raster.py

```diff
@@ -263,6 +267,58 @@ def get_data_at_coords(dataset: AmazonGeoTiff, x: float, y: float, month: int) -
   else:
     return value
 
+def get_predictions_at_each_pixel(
+    model: tf.keras.Model,
+    feature_transformer: ColumnTransformer,
+    geotiffs: dict[str, AmazonGeoTiff],
+    bounds: Bounds):
+
+  # Initialize a blank plane representing means and variance.
+  predicted_np = np.ma.array(
+      np.zeros([bounds.raster_size_x, bounds.raster_size_y, 2], dtype=float),
+      mask=np.ones([bounds.raster_size_x, bounds.raster_size_y, 2], dtype=bool))
+
+  for x_idx, x in enumerate(tqdm(np.arange(bounds.minx, bounds.maxx, bounds.pixel_size_x, dtype=float))):
+    rows = []
+    row_indexes = []
+    for y_idx, y in enumerate(np.arange(bounds.miny, bounds.maxy, -bounds.pixel_size_y, dtype=float)):
+      # Row should contain all the features needed to predict, in the same
+      # column order the model was trained.
+      row = {}
+      row["lat"] = y
+      row["long"] = x
+
+      # Surround in try/except as we will be trying to fetch out of bounds data.
+      try:
+        for geotiff_label, geotiff in geotiffs.items():
+          row[geotiff_label] = geotiff.value_at(x, y)
+          if pd.isnull(row[geotiff_label]):
+            raise ValueError
+      except (ValueError, IndexError):
+        continue # masked and out-of-bounds coordinates
```

Comment:
you'll want to at least print out an error message indicating this has occurred.

Reply:
Actually, this exception is used for coords outside the bounds of brazil unfortunately. The print statement would be really noisy.

Also, there shouldn't be an exception here.